### PR TITLE
Skip tasks in html comments

### PR DESCRIPTION
**Pull request by jmneri-shopify**
Added custom detection for html comments plus a unit test to confirm it's working as expected.

### CODE REVIEW

File: utils/extract.js [hunk 1]
```diff
@@ -15,12 +15,38 @@ function checklistItems(body) {
     const githubFlavoredMarkdown = true
     const tokens = marked.lexer(body, {gfm: githubFlavoredMarkdown})
 
-    /** @type {ChecklistItem[]} */
-    let items = []
+    return unroll(tokens).flatMap(token => checkableItem(token))
```
[the remark this answers is not in the document]
Comment (ashfurrow) [reply utils/extract.js:18]: Much nicer 💆

File: utils/extract.js [hunk 2]
```diff
@@ -15,12 +15,38 @@ function checklistItems(body) {
...
+}
+
+/**
+ * @param {marked.TokensList} tokens
+ * @returns {marked.Token[]}
+ */
+function unroll(tokens) {
```
[the remark this answers is not in the document]
Comment (ashfurrow) [reply utils/extract.js:25]: We should add this description as documentation to the function 👍

File: utils/tests/extract.test.js [hunk 3]
```diff
@@ -33,12 +42,44 @@ describe('extract', () => {
...
                 {text: 'Inflated checklist item', checked: true},
             ])
         })
+        it('should exclude items inside html comments', () => {
+            const body = "<!-- - [ ] Single line comment -->\n" +
+                        "- [ ] Visible checklist item\n" +
+                        "<!--\n" +
+                        "- [ ] Commented out checklist item\n" +
+                        "-->\n" +
+                        "Midline block comment: <!--\n" +
```
[the remark this answers is not in the document]
Comment (ashfurrow) [reply utils/tests/extract.test.js:61]: Yes, good call including a test case 👍

**Review by ashfurrow — APPROVED** [review]
Looks good, just a suggestion to add a PR comment to the JSDoc 👍 Thanks JM!

File: utils/extract.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmneri-shopify) [reply utils/extract.js:25]: Fair point, I should give it a less generic function name I think.

**Pull request merged**